diagnostic.md

```diff
@@ -3,32 +3,41 @@
 What are three advantages of using Ember?
 
 ```sh
-# your answer here
+Routability
+Less AJAX
+Responsiveness - The front end does not need to make as many requests to the back
+end for data it needs
 ```
 
 What is the templating system used in Ember? How it is related to
 handlebars?
 
 ```sh
-# your answer here
+The templating system is a way of setting the HTML content in the route or component.
+It works by writing handlebars to access different routes or components. Templates can
+also reference different components by name
 ```
 
 Name at least two layers of an Ember application (1.0 or 2.0):
 
 ```sh
-# your answer here
+Routes and Models
 ```
 
 How does a component differ from an Ember template?
 
 ```sh
-# your answer here
+A component does not have access to the entire route, instead its scope is defined at
+the location where the Component is invoked. Making it very re-usable
 ```
 
 What is an opinionated framework? Is Ember opinionated? How or how not?
 
 ```sh
-# your answer here
+An opinionated framework is a framework that forces you to write and do things in its
+own way. Ember is very opinionated simply for the reason that if you do not follow
+the strict guidelines it has, many things of your code will not work. For example, using
+either kebab-case or camel-case for naming files.
 ```
 
 In Ember what is a route's job (please list three)?
```

Comment:
1. Parse information from the url
2. Link the route to a component or template
3. Load the template/UI data via the 'model' hook.